pkg/bots/tgbot/utils.go

```diff
@@ -82,11 +86,51 @@ func FullNameFromFirstAndLastName(firstName, lastName string) string {
 }
 
 func ExtractTextFromMessage(message *tgbotapi.Message) string {
-	if message.Caption != "" {
-		return message.Caption
+	text := lo.Ternary(message.Caption != "", message.Caption, message.Text)
+
+	type MarkdownLink struct {
+		Markdown []uint16
+		Start    int
+		End      int
+	}
+
+	textUtf16 := utf16.Encode([]rune(text))
+	links := lop.Map(message.Entities, func(entity tgbotapi.MessageEntity, i int) MarkdownLink {
+		startIndex := entity.Offset
+		endIndex := startIndex + entity.Length
+		var title string
+		var href string
+		if entity.Type == "url" {
+			href = string(utf16.Decode(textUtf16[startIndex:endIndex]))
+			result, err := LinkPreview.PreviewLink(href, nil)
```

Comment:
This err should be logged, but this err doesnt really matter... How about adding an err field to MarkdownLink to handle the outside functions?

Reply:
Many sites does not have a preview (say, Twitter) and doing so may make the code hard to maintain. Maybe we can move this function to `chat_histories.go`.

Reply:
Sure.